### PR TITLE
Update changelog for 2.0.0 release

### PR DESCRIPTION
**Pull request by iambmelt**
Opening this PR to update the changelog -- a subsequent branch change will be made to rename the following:
```
release/1.16.5 -> release/2.0.0
```

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,5 +1,20 @@
+Version 2.0.0
```
Comment (iambmelt) [inline changelog.txt]: @rpdome Anything I am forgetting here?

File: README.md [hunk 2]
```diff
@@ -19,7 +19,13 @@ ADAL for Android gives you the ability to add support for Work Accounts to your
 
 A Work Account is an identity you use to get work done from your organization or school. Anywhere you need to get access to your work life you'll use a Work Account. The Work Account can be tied to an Active Directory server running in your datacenter or live completely in the cloud like when you use Office 365. A Work Account will be how your users know that they are accessing their important documents and data backed my Microsoft security.
 
-## ADAL for Android 1.16.3 Released!
+## ADAL for Android 2.0.0 Released!
+>Please note: ADAL 2.0.0 (released March 2020) is neither API-compatible nor cache-compatible the following versions:<br/>
```
Comment (shahzaibj) [inline README.md]: add `with` after cache-compatible

File: changelog.txt [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+-------------
+- Please note: ADAL 2.0.0 (released March 2020) is neither API-compatible nor cache-compatible the following versions:
```
Comment (shahzaibj) [inline changelog.txt]: I remember hearing something like going straight to 3.0.0 Is that no longer happening?

File: changelog.txt [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+    * 2.0-alpha   (released 2015-07-27)
+    * 2.0.1-alpha (released 2015-09-25)
+    * 2.0.2-alpha (released 2016-05-27)
+    * 2.0.3-alpha (released 2016-06-11)
+    * 2.0.4-alpha (released 2017-02-17)
+- First public release with AndroidX support
+- Bugfix: Resolved ADAL/1498
+    * Fixed an issue whereby tokens from a sovereign cloud environment would result in cache-misses
+- Fixes CVE-2019-17195
+    * Connect2id Nimbus JOSE+JWT before v7.9 can throw various uncaught exceptions while parsing a JWT, which could result in an application crash (potential information disclosure) or a potential authentication bypass.
```
Comment (rpdome) [inline changelog.txt]: wasn't this fixed in the previous release/hotfix already?

File: changelog.txt (hunk 3 above)
Comment (iambmelt) [reply changelog.txt]: We'll stick with `2.0.0`

File: README.md (hunk 2 above)
Comment (iambmelt) [reply README.md]: Good catch, will do

File: changelog.txt (hunk 4 above)
Comment (iambmelt) [reply changelog.txt]: [Yeah, you're right](https://repo1.maven.org/maven2/com/microsoft/aad/adal/1.16.3-hf2/adal-1.16.3-hf2.pom)

File: README.md (hunk 2 above)
Comment (iambmelt) [reply README.md]: Corrected in 12696342ce21a103ab129ade131db976b5ccc2e0

File: changelog.txt (hunk 4 above)
Comment (iambmelt) [reply changelog.txt]: Removed in 12696342ce21a103ab129ade131db976b5ccc2e0

**Pull request merged**